mcs/class/corlib/System.IO/FileStream.cs

```diff
@@ -188,6 +188,12 @@ internal FileStream (string path, FileMode mode, FileAccess access, FileShare sh
 				throw new ArgumentException ("Name has invalid chars");
 			}
 
+			// IsolatedStorage needs to keep the Name property to the default "[Unknown]"
+			if (!anonymous) {
+				path = Path.InsecureGetFullPath (path);
```

Comment:
Because anonymous does not store path and it was moved up to throw correct value in exceptions as well